crates/torii/graphql/src/query/mod.rs

```diff
@@ -95,6 +95,16 @@ fn parse_nested_type(
     TypeData::Nested((TypeRef::named(target_type), nested_mapping))
 }
 
+fn remove_trailing_zeros(value: Value) -> Value {
+    if let Value::String(str_val) = &value {
+        let hex_part = str_val.trim_start_matches("0x");
```

Comment:
Let's ensure that the string starts with `0x` before stripping

Reply:
Added!